src/main/java/de/caritas/cob/agencyservice/api/repository/agency/AgencyTenantSupportRepository.java

```diff
@@ -0,0 +1,42 @@
+package de.caritas.cob.agencyservice.api.repository.agency;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression;
+import org.springframework.context.annotation.Primary;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+/**
+ * Repository for the Agency entity in tenant context
+ */
+@Primary
+@ConditionalOnExpression("${multitenancy.enabled:true}")
+public interface AgencyTenantSupportRepository extends AgencyRepository {
```

Comment:
I would prefer to name classes TenantAware instead of TenantSupport

Reply:
changed